tempest/api/workloadmgr/barbican/test_barbican.py

```diff
@@ -3120,3 +3134,215 @@ def test_10_barbican(self):
+            trust_list = self.get_trusts()
+            found = False
+            for trust in trust_list:
+                if trust['name'] == self.wlm_trust['name']:
+                    found = True
+                    break
+            if found and self.wlm_trust_valid and self.wlm_scheduler:
+                reporting.add_test_step("Verify valid trust", tvaultconf.PASS)
+            else:
+                reporting.add_test_step("Verify valid trust", tvaultconf.FAIL)
+                reporting.set_test_script_status(tvaultconf.FAIL)
+
+        except Exception as e:
+            reporting.add_test_step(str(e), tvaultconf.FAIL)
+            reporting.set_test_script_status(tvaultconf.FAIL)
+        finally:
+            reporting.test_case_to_write()
+
+    @decorators.attr(type='workloadmgr_cli')
+    def test_12_barbican(self):
+        try:
+            reporting.add_test_script(str(__name__) + \
+                    "_validate_scheduler_trust_with_scheduler_disabled")
+            global vm_id
+            global volume_id
+            global exception
+            global secret_uuid
+            if exception != "":
+                LOG.error("pre req failed")
+                raise Exception(str(exception))
+            LOG.debug("pre req completed")
+
+            # Create workload with scheduler disabled using CLI
+            workload_create = \
+                    command_argument_string.workload_create_with_encryption +\
+                " --instance instance-id=" + \
+                str(vm_id) + " --jobschedule enabled=False" + \
+                " --secret-uuid " + str(secret_uuid)
+            rc = cli_parser.cli_returncode(workload_create)
+            if rc != 0:
+                reporting.add_test_step(
+                    "Execute workload-create command with scheduler disable",
+                    tvaultconf.FAIL)
+                raise Exception(
+                    "Command workload create did not execute correctly")
+            else:
+                reporting.add_test_step(
+                    "Execute workload-create command with scheduler disable",
+                    tvaultconf.PASS)
+
+            time.sleep(10)
+            self.wid = query_data.get_workload_id_in_creation(tvaultconf.workload_name)
+            LOG.debug("Workload ID: " + str(self.wid))
+            if(self.wid is not None):
+                self.wait_for_workload_tobe_available(self.wid)
+                if(self.getWorkloadStatus(self.wid) == "available"):
+                    reporting.add_test_step(
+                        "Create workload with scheduler disable", tvaultconf.PASS)
+                else:
+                    reporting.add_test_step(
+                        "Create workload with scheduler disable", tvaultconf.FAIL)
+                    reporting.set_test_script_status(tvaultconf.FAIL)
+            else:
+                raise Exception("Create workload with scheduler disabled")
```

Comment:
Are we not going to set test step to FAIL in case of failure in workload creation ?

Reply:
In except block, we are adding the test step with FAIL status as well as setting the test script status to FAIL. Hence just raising the exception, takes care of it automatically.